### ISSUE TITLE
Delete entries are not generated for removed files in some cases

### ISSUE THREAD

**Issue by busykai** [clearlinux/mixer-tools#564]
When generating a downstream version, mixer fails to generate delete entries in the bundle manifest under the conditions described in details below.

First of all, the simplest case of updating an RPM with some files removed from it, the delete entries are properly generated:
```
Manifest.os-core:.d..   0000000000000000000000000000000000000000000000000000000000000000        20      /usr/lib/systemd/system/multi-user.target.wants/systemd-networkd.service
Manifest.os-core:.d..   0000000000000000000000000000000000000000000000000000000000000000        20      /usr/lib/systemd/system/sockets.target.wants/systemd-networkd.socket
```
However, when I add an unrelated bundle, edit it, and then re-generate a new version (with --clean), I don't get the delete entries in the manifests.

Specific steps are:
```
$ mixer versions update
$ mixer bundle add NetworkManager
$ mixer bundle edit NetworkManager
$ (edit the bundle to remove some deps, e.g. libX11client)
$ sudo mixer build all --native
```
After that, the Manifests are simply do not contain any mention of the removed files.

I also tested the following three scenarios:

1. (WORKS) If I add the unrelated existing upstream bundle to version 10 (as
   opposed to after upgrading to version 20), the entry does appear.

2. (WORKS) If I add a new bundle (I was using NetworkManager-basics, containing, for example, just NetworkManager package), the delete entry appears.

3. (DOES NOT WORK) If add a new bundle (NetworkManager-basics) and include it in
   NetworkManager (replacing the corresponding entries), the delete entry does
   not appear.

4. (DOES NOT WORK) While trying to go back to 2, I reverted the local copy of
   NetworkManager bundle to be like upstream, but local, the delete entry does
   not appear.

5. (DOES NOT WORK) So I removed local bundle and added pristine upstream again,
   rebuilt, the entry did not appear. Which is weird, because technically, the
   state is like 2.

I always used the same version 10 and was rebuilding the version 20 using "sudo mixer build all --native --clean".

**Comment by jwakre** [comment]
This looks like an issue with the **'--clean'** flag. When rebuilding a version with the --clean flag, the 'previous version' field in the header is incorrect along with other manifest information, like file deletes.

These errors are caused because the --clean flag does not revert the previous version file (update/image/LAST_VER). After manually changing the LAST_VER file, I could successfully rebuild a mix with the --clean flag.

**Comment by busykai** [comment]
@jwakre, yeah. it looks like it is `--clean` issue... i thought it would help me: i was not certain about side-effects a rebuilding version without `--clean` would have, but it backfired, obviously. thanks for looking into this!

**Comment by jwakre** [comment]
The clean issue is fixed by #565
